browser/src/slideshow/PerformTransition.ts

```diff
@@ -114,24 +114,17 @@ SlideShow.PerformTransition = function (
 	let subTypeIndex = 1;
 	const transitionSubType =
 		stringToTransitionSubTypeMap[slideInfo.transitionSubtype];
+	const transitionDirection = slideInfo.transitionDirection;
+
 	switch (stringToTransitionTypeMap[slideInfo.transitionType]) {
 		case TransitionType.FADE:
-			if (
-				slideInfo.transitionSubtype &&
-				slideInfo.transitionSubtype.length > 0
-			) {
-				if (transitionSubType == TransitionSubType.CROSSFADE) {
-					subTypeIndex = 2;
-				} else if (
-					transitionSubType == TransitionSubType.FADEOVERCOLOR &&
-					slideInfo.transitionDirection
-				) {
-					subTypeIndex = 1;
-				} else {
-					subTypeIndex = 0;
-				}
-			}
-			new SlideShow.FadeTransition(canvas, image1, image2).start(subTypeIndex);
```

Comment:
can we have just `slideInfo` passed into transition itself?

`new SlideShow.FadeTransition(canvas, image1, image2, slideInfo).start()`

then every transition will be able to decide inside what to do with it